include/oneapi/dpl/internal/distributed_ranges_impl/detail/sycl_utils.hpp

```diff
@@ -0,0 +1,258 @@
+    }
+    else if constexpr (Dim == 3)
+    {
+        sycl::range local(8, 8, 8);
+        return sycl::nd_range<Dim>(
+            sycl::range(round_up(global[0], local[0]), round_up(global[1], local[1]), round_up(global[2], local[2])),
+            local);
+    }
+    else
+    {
+        assert(false);
+        return sycl::range(0);
+    }
+}
+
+template <typename Fn>
+sycl::event
+parallel_for_nd(sycl::queue& q, sycl::range<1> global, Fn&& fn)
+{
+    return q.parallel_for(nd_range(global), [=](auto nd_idx) {
+        auto idx0 = nd_idx.get_global_id(0);
+        if (idx0 < global[0])
+        {
+            fn(idx0);
+        }
+    });
+}
+
+template <typename Fn>
+sycl::event
+parallel_for_nd(sycl::queue& q, sycl::range<2> global, Fn&& fn)
+{
+    return q.parallel_for(nd_range(global), [=](auto nd_idx) {
+        auto idx0 = nd_idx.get_global_id(0);
+        auto idx1 = nd_idx.get_global_id(1);
+        if (idx0 < global[0] && idx1 < global[1])
+        {
+            fn(std::array{idx0, idx1});
+        }
+    });
+}
+
+template <typename Fn>
+sycl::event
+parallel_for_nd(sycl::queue& q, sycl::range<3> global, Fn&& fn)
+{
+    return q.parallel_for(nd_range(global), [=](auto nd_idx) {
+        auto idx0 = nd_idx.get_global_id(0);
+        auto idx1 = nd_idx.get_global_id(1);
+        auto idx2 = nd_idx.get_global_id(2);
+        if (idx0 < global[0] && idx1 < global[1] && idx2 < global[2])
+        {
+            fn(std::array{idx0, idx1, idx2});
+        }
+    });
+}
+
+auto
+combine_events(sycl::queue& q, const auto& events)
+{
+    return q.submit([&](auto&& h) {
+        h.depends_on(events);
+        // Empty host task necessary due to https://github.com/oneapi-src/distributed-ranges/issues/778
+        h.host_task([] {});
+    });
+}
+
+template <typename Fn>
+sycl::event
+parallel_for(sycl::queue& q, sycl::range<1> numWorkItems, Fn&& fn)
+{
+    std::size_t block_size = 128;
+    std::size_t max_kernel_size = max_kernel_size_();
+
+    if (numWorkItems.size() < max_kernel_size)
+    {
+        return parallel_for_workaround(q, numWorkItems, std::forward<Fn>(fn), block_size);
+    }
+    else
+    {
+        return parallel_for_64bit(q, numWorkItems, std::forward<Fn>(fn));
+    }
+}
+
+template <typename Fn>
+sycl::event
+parallel_for(sycl::queue& q, sycl::range<2> global, Fn&& fn)
+{
+    auto max = std::numeric_limits<std::int32_t>::max();
+    assert(global[0] < max && global[1] < max);
+    return parallel_for_nd(q, global, fn);
+}
+
+template <typename Fn>
+sycl::event
+parallel_for(sycl::queue& q, sycl::range<3> global, Fn&& fn)
+{
+    auto max = std::numeric_limits<std::int32_t>::max();
+    assert(global[0] < max && global[1] < max && global[2] < max);
+    return parallel_for_nd(q, global, fn);
+}
+
+using event = sycl::event;
+
+} // namespace oneapi::dpl::experimental::dr::__detail
+
+#else
+
+namespace oneapi::dpl::experimental::dr::__detail
+{
+
+class event
```

Comment:
What is the purpose of this (no-op) event class?

Reply:
This class is only used (as a dummy event class) in the multi-process model, which is not included in this PR.

I'll let @lslusarczyk chime in on whether to keep it, but I just submitted a PR removing it #1718.

Reply:
Thank you for PR. Let's don't keep any dead code. Merged.